### ISSUE TITLE
Renault Megane Mk3 Rlink - Vin write

### ISSUE THREAD

**Issue by Ghighi246** [cedricp/ddt4all#470]
Hello,

I have a renault megane mk3 with rlink retrofitted. When I check the settings in MFD ECU I saw that the VIN number is not inserted (there are a lot of 0).

Can I simply write the VIN number with ddt4all or EcuTweaker (Android)? Would it brick something?

Thanks in advance!

**Comment by BarektE** [comment]
ELM327 CAN 2 portunu destekliyorsa VIN No: yazabilirsin.  Normalde ELM327 de Modifikasyon yapmanız gerekir.

**Comment by Ghighi246** [comment]
I have a modified extender in order to access Multimedia CAN. My problem was that: can I brick the device if I write this VIN number?

**Comment by fparmentier** [comment]
Yes you can write the VIN but there is not really a need to do so as far as I know.

**Comment by Ghighi246** [comment]
Thank you all for your answers! Really helpful.

I want to write the VIN number because mai ecodrive function is not making any graphic calculations. For the moment the bord computer is shown correctly (distantce, consumption etc.) but the leaf is at 80/100 all the time, acceleration and anticipation graph is full and gear change is null. I don't understand what is set wrong on it.

On another forum I seen that someone mentioned that the crc number is used to make different ecu communicate accordingly. So I hope this crc number and vin write will solve my problema.

**Comment by fparmentier** [comment]
Sounds interesting. I have also retrofitted RLink and don't have Eco drive working. Will change the VIN to see if I can make it work.

**Comment by Ghighi246** [comment]
But the bord computer is transfering it's data from meter cluster to mfd?

**Comment by fparmentier** [comment]
I have no idea

**Comment by Ghighi246** [comment]
You can activate the ecodrive function from mfd settings with ddt4all. But if you have meter cluster before 2012 it will not work for sure

**Comment by fparmentier** [comment]
Thanks, mine is from 2010. So I know what I have to do, buy a new cluster.

**Comment by Ghighi246** [comment]
So, i managed to write cid but no update. Eco function still not calculate the acc gear and anticipation. Could be the bic?

**Comment by fparmentier** [comment]
Maybe you have an old version of the BIC? I had do upgrade my BIC to make RS Monitor work, could be the same for eco functions?

**Comment by Ghighi246** [comment]
Hy, when I replaced BIC I installed one with the following number 283467680R. When I search online I find this BIC as dismantled from Clio IV or Captur.

**Comment by fparmentier** [comment]
I managed to find a new cluster (2012 analog). I installed it, changed the milage to match my car, did a rollback of the original configuratiton (PyRen) and changed the VIN in cluster and Mfd. Result: Eco functions not working.

**Comment by Ghighi246** [comment]
Hy, what do you mean by analog? It was an RS version? Or a version from fluence (megane 3 from turkey)? 
Exactly what are you doing I want to do next week by borrowing a cluster from a friend. He has a higher mileage then me and I don't know how to proceed in order not to write his mileage into my abs ecu.

**Comment by fparmentier** [comment]
As long as you don't drive with the car, the ABS and ECU won't get updated mileage (as far as I know!)
It was an GT version, I drive a Megane GT 180 2010 model and changed the cluster with a 2012 model. I updated the mileage by reading the eeprom from both clusters and copied the values from my old cluster onto my new cluster. After that wrote eeprom back. There are several ways to read eeprom, I did it with (genuine) Renolink.

**Comment by Ghighi246** [comment]
Ok. I don't have renolink, just an elm327. I think I will try first with a different bic to see if there is anything changed... Then I will try with different cluster.
Maybe your cluster was on the limit with software version inside... Also, if your megane is with sport cluster, maybe you should set the bic to different script, for x95 RS

**Comment by Furtif** [comment]
https://github.com/cedricp/ddt4all/issues/836
